In [2]:
import pandas as pd
import requests
from io import StringIO

# Criação do dataFrame dos alunos
# ID do arquivo no Google Drive
file_id = '1inLJ0QY0vOhD_2CKujdLEUvF4MnPbAi5'

# URL modificada para forçar o download do arquivo
url = f"https://drive.google.com/uc?id={file_id}"

# Tentando obter o arquivo com requests
try:
    response = requests.get(url)
    response.raise_for_status()  # Lança um erro para respostas não-sucedidas
    # Usando StringIO para converter o texto em um arquivo em memória e, então, lendo com o Pandas
    csv_raw = StringIO(response.text)
    ecom = pd.read_csv(csv_raw)
except requests.RequestException as e:
    print(f"Erro ao acessar o arquivo: {e}")

In [3]:
ecom

ID corredor_armazem modo_envio  numero_chamadas_cliente  \
0          1                D      Aviao                        4   
1          2                F      Aviao                        4   
2          3                A      Aviao                        2   
3          4                B      Aviao                        3   
4          5                C      Aviao                        2   
...      ...              ...        ...                      ...   
10995  10996                B      Navio                        4   
10996  10997                C      Navio                        5   
10997  10998                F      Navio                        5   
10998  10999                D      Navio                        2   
10999  11000                C      Navio                        4   

       avaliacao_cliente  custo_produto  compras_anteriores  \
0                      2            177                   3   
1                      5            216                   2   
2                      2            183                   4   
3                      3            176                   4   
4                      2            184                   3   
...                  ...            ...                 ...   
10995                  1            232                   5   
10996                  4            242                   5   
10997                  2            223                   6   
10998                  5            155                   5   
10999                  2            255                   5   

      prioridade_produto genero  desconto  peso_gramas  entregue_no_prazo  
0                  baixa      F        44         1233                  1  
1                  baixa      M        59         3088                  1  
2                  baixa      M        48         3374                  1  
3                  media      M        10         1177                  1  
4                  media      F        46         2484                  1  
...                  ...    ...       ...          ...                ...  
10995              media      F         6         1247                  0  
10996              baixa      F         4         1155                  0  
10997              media      M         2         1210                  0  
10998              baixa      F         6         1639                  0  
10999              baixa      F         6         1739                  0  

[11000 rows x 12 columns]

In [4]:
ecom.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11000 entries, 0 to 10999
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   ID                       11000 non-null  int64 
 1   corredor_armazem         11000 non-null  object
 2   modo_envio               11000 non-null  object
 3   numero_chamadas_cliente  11000 non-null  int64 
 4   avaliacao_cliente        11000 non-null  int64 
 5   custo_produto            11000 non-null  int64 
 6   compras_anteriores       11000 non-null  int64 
 7   prioridade_produto       11000 non-null  object
 8   genero                   11000 non-null  object
 9   desconto                 11000 non-null  int64 
 10  peso_gramas              11000 non-null  int64 
 11  entregue_no_prazo        11000 non-null  int64 
dtypes: int64(8), object(4)
memory usage: 1.0+ MB


## Análise Exploratória

### Separação de Variáveis numéricas das variáveis categóricas

In [5]:
nums = ['numero_chamadas_cliente', 'custo_produto', 'compras_anteriores', 'desconto', 'peso_gramas', 'entregue_no_prazo']

In [6]:
categ = ['corredor_armazem', 'modo_envio', 'avaliacao_cliente', 'prioridade_produto', 'genero']

In [7]:
target = ['entregue_no_prazo']

In [8]:
ecom[nums].describe()

numero_chamadas_cliente  custo_produto  compras_anteriores  \
count             11000.000000   11000.000000        11000.000000   
mean                  4.054455     210.200909            3.567727   
std                   1.141438      48.062985            1.522852   
min                   2.000000      96.000000            2.000000   
25%                   3.000000     169.000000            3.000000   
50%                   4.000000     214.000000            3.000000   
75%                   5.000000     251.000000            4.000000   
max                   7.000000     310.000000           10.000000   

           desconto   peso_gramas  entregue_no_prazo  
count  11000.000000  11000.000000       11000.000000  
mean      13.372545   3633.844455           0.596636  
std       16.204943   1635.402722           0.490595  
min        1.000000   1001.000000           0.000000  
25%        4.000000   1839.000000           0.000000  
50%        7.000000   4149.000000           1.000000  
75%       10.000000   5050.000000           1.000000  
max       65.000000   7846.000000           1.000000

In [9]:
for coluna in categ:
    print(f"Contagem de valores para {coluna}:")
    print(ecom[coluna].value_counts())
    print("\n")

Contagem de valores para corredor_armazem:
corredor_armazem
F    3666
D    1834
C    1834
A    1833
B    1833
Name: count, dtype: int64


Contagem de valores para modo_envio:
modo_envio
Navio       7463
Aviao       1777
Caminhao    1760
Name: count, dtype: int64


Contagem de valores para avaliacao_cliente:
avaliacao_cliente
3    2239
1    2235
4    2189
5    2171
2    2166
Name: count, dtype: int64


Contagem de valores para prioridade_produto:
prioridade_produto
baixa    5298
media    4754
alta      948
Name: count, dtype: int64


Contagem de valores para genero:
genero
F    5546
M    5454
Name: count, dtype: int64




In [10]:
ecom[target].value_counts()

entregue_no_prazo
1                    6563
0                    4437
Name: count, dtype: int64

## Limpeza de Dados

Vamos começar checando os Null e NaN.

In [11]:
ecom.isnull().sum()

ID                         0
corredor_armazem           0
modo_envio                 0
numero_chamadas_cliente    0
avaliacao_cliente          0
custo_produto              0
compras_anteriores         0
prioridade_produto         0
genero                     0
desconto                   0
peso_gramas                0
entregue_no_prazo          0
dtype: int64

In [12]:
ecom.isna().sum()

ID                         0
corredor_armazem           0
modo_envio                 0
numero_chamadas_cliente    0
avaliacao_cliente          0
custo_produto              0
compras_anteriores         0
prioridade_produto         0
genero                     0
desconto                   0
peso_gramas                0
entregue_no_prazo          0
dtype: int64

Não existem valores ausentes e 'Not a Number' no conjunto de dados, mas será que nossos problemas estão resolvidos ? 

Também podemos checar registros duplicados.

In [26]:
ecom.duplicated().sum()

0

### Estudo de Outliers

Outliers são pontos de dados que se distanciam significativamente dos demais em um conjunto, representando valores incomuns. Esses valores atípicos podem afetar análises estatísticas, possivelmente ocultando descobertas importantes ou distorcendo resultados. A identificação de outliers não segue regras estatísticas rígidas, dependendo mais do conhecimento específico da área e da análise do processo de coleta de dados. Embora não haja uma definição matemática precisa, métodos e testes estatísticos ajudam a detectar esses pontos discrepantes, que são cruciais por seu potencial de impactar análises estatísticas adversamente.

Vamos começar a análise olhando para as variáveis numéricas

In [27]:
ecom[nums].describe()

numero_chamadas_cliente  custo_produto  compras_anteriores  \
count             11000.000000   11000.000000        11000.000000   
mean                  4.054455     210.200909            3.567727   
std                   1.141438      48.062985            1.522852   
min                   2.000000      96.000000            2.000000   
25%                   3.000000     169.000000            3.000000   
50%                   4.000000     214.000000            3.000000   
75%                   5.000000     251.000000            4.000000   
max                   7.000000     310.000000           10.000000   

           desconto   peso_gramas  entregue_no_prazo  
count  11000.000000  11000.000000       11000.000000  
mean      13.372545   3633.844455           0.596636  
std       16.204943   1635.402722           0.490595  
min        1.000000   1001.000000           0.000000  
25%        4.000000   1839.000000           0.000000  
50%        7.000000   4149.000000           1.000000  
75%       10.000000   5050.000000           1.000000  
max       65.000000   7846.000000           1.000000

O atributo desconto está com características interessantes:

 - a média(mean): 13.372545
 - o desvio-padrão(std): 16.204943
 - o menor desconto (min): 1.000000
 - o maior desconto (max): 65.00000

 Além disso 75% dos produtos estão com desconto menor do que 10%, ou seja, a maioria dos produtos está com desconto de até 10%, assim podemos concluir que descontos muito maiores que 10% podem sem outliers.

In [29]:
ecom.desconto.describe()

count    11000.000000
mean        13.372545
std         16.204943
min          1.000000
25%          4.000000
50%          7.000000
75%         10.000000
max         65.000000
Name: desconto, dtype: float64

Precisamos de métodos estatísticos para analisar os OutLiers...